pkg/template/admission/impersonation/admission.go

```diff
@@ -0,0 +1,68 @@
+package impersonation
+
+import (
+	"errors"
+	"io"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	"github.com/openshift/origin/pkg/client"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+	userapi "github.com/openshift/origin/pkg/user/api"
+	"k8s.io/kubernetes/pkg/admission"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+)
+
+func init() {
+	admission.RegisterPlugin("openshift.io/TemplateImpersonationAdmissionPlugin", func(c clientset.Interface, config io.Reader) (admission.Interface, error) {
+		return &impersonationAdmissionHandler{
+			Handler: admission.NewHandler(admission.Create, admission.Update),
+		}, nil
+	})
+}
+
+type impersonationAdmissionHandler struct {
+	*admission.Handler
+	oc client.Interface
+}
+
+func (iah *impersonationAdmissionHandler) Admit(attr admission.Attributes) (err error) {
```

Comment:
This started out in the Validate function, but I understood @deads2k didn't want me to make the function impure.  He suggested making an admission controller.  Is the answer to stick with an admission controller or move these to the strategy (outside the Validate call?)